static_frame/test/unit/test_index.py

```diff
@@ -1644,6 +1649,13 @@ def test_index_iloc_map_b(self) -> None:
         with self.assertRaises(KeyError):
             ih1.iter_label().apply(ih2._loc_to_iloc)
 
+    def test_index_iloc_map_c(self) -> None:
+        ih1 = Index(['a', np.nan])
+        ih2 = Index([np.nan, 'a'])
+
+        post = ih1._index_iloc_map(ih2)
+        self.assertEqual(post.tolist(), [1, 0])  # What should this be?
```

Comment:
@flexatone What do you think this behavior should be? I'm really unfamiliar with the rules regarding support for `Index` objects with `nan`s

Reply:
NumPy's `np.nan` as single instance of a NaN value that can be used to match object identity:
```
>>> np.nan is np.nan
True
>>> np.nan is (np.nan+1)
False
```
Of course, normally we would not expect NaNs to have the same object id. But by using the np.nan, we actually get a Python object that can be reliably used in a dictionary and SF index, but only from that very same np.nan instance:
```
>>> d = {np.nan: 0, None: 1}
>>> d[np.nan]
0
>>> d[float('nan')]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: nan
```
So indeed, in the example you provide, that np.nans can map to each other just as if they were matching numbers. Of course, this will not work with any NaN. 

But in general, SF's index objects treat NaNs just as NaNs are treated in a dictionary, which may still be surprising, but is consistent.
```
>>> d = {np.nan: 0, None: 1, float('nan'): 2, float('nan'): 3}
>>> d[np.nan]
0
>>> d[float('nan')]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: nan
>>> d[list(d.keys())[-1]]
3
```